### PR TITLE
[image-plugin] feat: updated settings to have expand and download toggle

### PR DESCRIPTION
**Pull request by tamrat-bay**

### CODE REVIEW

File: packages/plugin-image/web/src/toolbar/image-settings.js [hunk 1]
```diff
@@ -42,6 +48,42 @@ class ImageSettings extends Component {
...
   }
 
+  toggleState = key => () => {
+    this.setState(prevState => ({
+      [key]: !prevState[key],
+    }));
+  };
+
+  renderToggle = ({ toggleKey, labelKey, dataHook }) => {
+    const tooltipText = toggleKey === 'isRightClickEnabled' ? this.imgCanBeDownloadedTooltip : null;
```
Comment (tamrat-bay) [inline packages/plugin-image/web/src/toolbar/image-settings.js:58]: move tooltip to data

File: e2e/tests/media.e2e.js [hunk 2]
```diff
@@ -92,9 +94,19 @@ describe('plugins', () => {
       cy.eyesCheckWindow(this.test.title + '  - plugin full width size');
     });
 
-    it('render image with link', () => {
-      cy.loadRicosEditorAndViewer('image-with-link');
-      cy.getImageLink();
+    it.only('should disable image expand', () => {
```
[the remark this answers is not in the document]
Comment (tamrat-bay) [reply e2e/tests/media.e2e.js:97]: done

File: packages/plugin-commons/web/statics/styles/input-with-label.scss [hunk 3]
```diff
@@ -39,8 +39,9 @@
 }
 
 .infoIcon {
-  vertical-align: text-bottom;
-  height: 17px;
+  margin-left: 4px;
```
[the remark this answers is not in the document]
Comment (tamrat-bay) [reply packages/plugin-commons/web/statics/styles/input-with-label.scss:42]: done

File: packages/editor-common/web/src/Icons/InfoIcon.tsx [hunk 4]
```diff
@@ -2,23 +2,12 @@
 import React from 'react';
 
 const InfoIcon = props => (
-  <svg width="19px" height="19px" viewBox="0 0 19 19" {...props}>
-    <defs>
+  <svg xmlns="http://www.w3.org/2000/svg" {...props} width={24} height={24} viewBox="0 0 24 24">
```
[the remark this answers is not in the document]
Comment (tamrat-bay) [reply packages/editor-common/web/src/Icons/InfoIcon.tsx:5]: done

File: packages/plugin-image/web/src/image-viewer.jsx [hunk 5]
```diff
@@ -322,7 +322,7 @@ class ImageViewer extends React.Component {
       e.stopPropagation(); // fix problem with wix platform, where it wouldn't scroll and sometimes jump to different page
       this.scrollToAnchor();
     } else {
-      this.handleExpand(e);
+      !this.props.componentData.config?.disableExpand && this.handleExpand(e);
```
[the remark this answers is not in the document]
Comment (tamrat-bay) [reply packages/plugin-image/web/src/image-viewer.jsx:325]: done

File: packages/plugin-image/web/src/image-viewer.tsx [hunk 6]
```diff
@@ -373,7 +374,17 @@ class ImageViewer extends React.Component<ImageViewerProps, ImageViewerState> {
...
+    const { componentData, disableRightClick } = this.props;
+    let disabled = false;
+    if (componentData.disableRightClick !== undefined) {
+      disabled = componentData.disableRightClick;
+    } else if (disableRightClick !== undefined) {
+      disabled = disableRightClick;
+    }
+    return disabled && e.preventDefault();
+  };
+  hasExpand = () => !this.props.componentData.config?.disableExpand && this.props.settings.onExpand;
```
Comment (tamrat-bay) [inline packages/plugin-image/web/src/image-viewer.tsx:387]: what about new line between functions

File: packages/plugin-image/web/src/toolbar/createToolbar.ts [hunk 7]
```diff
@@ -33,7 +33,7 @@ const createToolbar: CreatePluginToolbar = ({
       isMobile,
       settings,
     }),
-    InsertButtons: createInsertButtons({ t, settings }),
+    InsertButtons: createInsertButtons({ t, settings, uiSettings }),
```
Comment (tamrat-bay) [inline packages/plugin-image/web/src/toolbar/createToolbar.ts:36]:
```suggestion
    InsertButtons: createInsertButtons({ t, settings, disableRightClick }),
```

File: packages/plugin-image/web/src/toolbar/image-settings.js [hunk 8]
```diff
@@ -19,29 +20,66 @@ class ImageSettings extends Component {
...
+    this.setState(prevState => ({
+      [key]: !prevState[key],
+    }));
+  };
+
+  renderToggle = ({ toggleKey, labelKey, dataHook, tooltipText }) => {
+    return (
+      <div key={toggleKey} className={this.styles.imageSettings_toggleContainer}>
+        <LabeledToggle
+          style={{ paddingTop: 24 }}
```
Comment (tamrat-bay) [inline packages/plugin-image/web/src/toolbar/image-settings.js:57]: delete

File: packages/plugin-image/web/src/toolbar/image-settings.js [hunk 9]
```diff
@@ -19,29 +20,66 @@ class ImageSettings extends Component {
...
+          theme={this.props.theme}
+          checked={this.state[toggleKey]}
+          label={this.props.t(labelKey)}
+          onChange={this.toggleState(toggleKey)}
+          dataHook={dataHook}
+          tooltipText={tooltipText ? tooltipText : null}
```
Comment (tamrat-bay) [inline packages/plugin-image/web/src/toolbar/image-settings.js:63]: This should probably work
```suggestion
          tooltipText={tooltipText}
```

File: packages/ricos-content/web/src/normalization/normalizeInitialState.spec.ts [hunk 10]
```diff
@@ -747,4 +749,14 @@ describe('normalizeInitialState', () => {
       expect(actual.entityMap['1'].data.config.link).toEqual({ anchor: 'cjvg0' });
     });
   });
+
+  describe('disableRightClick normalizer', () => {
+    it('should add disableRightClick prop to image and gallery componentData', () => {
+      const actual = normalizeInitialState(imageGalleryContentState, {
```
Comment (tamrat-bay) [inline packages/ricos-content/web/src/normalization/normalizeInitialState.spec.ts:755]: beforeEach() {
deepClone imageGalleryCont...
}

File: packages/ricos-content/web/src/normalization/normalizeInitialState.spec.ts [hunk 11]
```diff
@@ -747,4 +750,41 @@ describe('normalizeInitialState', () => {
       expect(actual.entityMap['1'].data.config.link).toEqual({ anchor: 'cjvg0' });
     });
   });
+
+  describe('disableRightClick normalizer', () => {
+    let imageGalleryInitialState: DraftContent = imageGalleryContentState;
```
Comment (tamrat-bay) [inline packages/ricos-content/web/src/normalization/normalizeInitialState.spec.ts:755]:
```suggestion
    let imageGalleryInitialState: DraftContent;
```

File: website/docs/ricos/ricos-api.mdx [hunk 12]
```diff
@@ -108,10 +110,14 @@ mediaSettings?: {
 When `true`, media like videos will be paused.
 A possible use case is for pausing videos on tab switch.
 
-#### `disableRightClick`
+#### `disableRightClick` (deprecated) if used the same behavior will remain
```
Comment (shaulgo) [inline website/docs/ricos/ricos-api.mdx:113]:
```suggestion
#### `disableRightClick` (deprecated)
```

File: website/docs/ricos/ricos-api.mdx [hunk 13]
```diff
@@ -93,9 +93,11 @@ foregroundColor: 'black'; <br />
 The string values should be css color strings (e.g `'black'`, `'#2F2E2E'`, etc) <br />
 
 ```jsx
+`Note disableRightClick?: boolean; is deprecated`
```
Comment (shaulgo) [inline website/docs/ricos/ricos-api.mdx:96]: remove this line

File: website/docs/ricos/ricos-api.mdx [hunk 14]
```diff
@@ -108,10 +110,14 @@ mediaSettings?: {
...
 A possible use case is for pausing videos on tab switch.
 
-#### `disableRightClick`
+#### `disableRightClick` (deprecated) if used the same behavior will remain
 
 Disables the mouse right click functionality for the gallery, video and image plugins.
 
+#### `disableDownload`
+
+Disables the mouse right click functionality for the gallery, video and image plugins and also sets the initial download mode for custom videos.
```
Comment (shaulgo) [inline website/docs/ricos/ricos-api.mdx:119]:
```suggestion
Sets the initial value of disableDownload for image, gallery and video plugins. 
Images - disables right click. 
Video - removes the download option from context menu.
```

File: website/docs/plugins_api/PluginCustomization.md [hunk 15]
```diff
@@ -124,14 +124,15 @@ The `getMetadataUrl` should provide a Website metadata service endpoint URL for
 | setting key     | description                            | default value | is required? | editor/viewer config |
 | --------------- | -------------------------------------- | ------------- | ------------ | -------------------- |
 | `onExpand`      | called when the user expands an image  | none          | No           | viewer               |
-| `disableExpand` | disables expanding images & fullscreen | false         | No           | viewer               |
+| `disableExpand` (deprecated!) | disables expanding images & fullscreen | false         | No           | viewer (this option is deprecated! if used it will be supported and the same behavior will remain. note that now the initial value needs to be set in the editor settings)           |
```
Comment (shaulgo) [inline website/docs/plugins_api/PluginCustomization.md:127]: no need to explain deprecated

**Review by shaulgo — CHANGES_REQUESTED** [review]

**Pull request merged**